Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Revenue.hs

```diff
@@ -78,31 +86,33 @@ getCollectionHistory merchantShortId volunteerId place mbFrom mbTo = do
             _ -> (.id.getId) <$> volunteers
       findAllByVolunteerIds merchant.id relevantIds from_ to
   onlineCollections <- findAllByStatus merchant.id CLEARED from_ to
-  let offlineCollectionsRes = getCollectionSummary offlineCollections
-  let onlineCollectionsRes = getCollectionSummary onlineCollections
+  let offlineCollectionsRes = getCollectionSummary offlineCollections rangeBasis
+  let onlineCollectionsRes = getCollectionSummary onlineCollections rangeBasis
   pure $ Common.CollectionList onlineCollectionsRes offlineCollectionsRes
 
-getCollectionSummary :: [DriverFee] -> Common.CollectionListRes
-getCollectionSummary collections = do
+getCollectionSummary :: [DriverFee] -> KS.PeriodType -> Common.CollectionListRes
+getCollectionSummary collections granuality = do
   let totalCnt = length collections
       totalFeeCollected = sum $ map calcFee collections
       totalRides = sum $ map (.numRides) collections
       totalDrivers = length $ DL.nub (map driverId collections)
       collectionsTs =
         [ (calculateTotalFee tmp, listToMaybe [x | (_, Just x) <- tmp])
           | tmp <-
-              DL.groupBy (\(_, a) (_, b) -> a == b) $
+              DL.groupBy (\(_, a) (_, b) -> makeGranular a == makeGranular b) $
                 sortBy (comparing snd) [(calcFee fee, fee.collectedAt) | fee <- collections]
         ]
       numRidesTs =
         [ (calculateTotalRides tmp, listToMaybe [x | (_, Just x) <- tmp])
           | tmp <-
-              DL.groupBy (\(_, a) (_, b) -> a == b) $
+              DL.groupBy (\(_, a) (_, b) -> makeGranular a == makeGranular b) $
                 sortBy (comparing snd) [(fee.numRides, fee.collectedAt) | fee <- collections]
         ]
-      driversPmtTs = [(length tmp, listToMaybe tmp) | tmp <- DL.group $ DL.sort $ catMaybes [fee.collectedAt | fee <- collections]]
+      driversPmtTs = [(length tmp, listToMaybe (catMaybes tmp)) | tmp <- DL.groupBy (\a b -> makeGranular a == makeGranular b) $ DL.sort [fee.collectedAt | fee <- collections]]
   Common.CollectionListRes totalCnt totalFeeCollected totalRides totalDrivers collectionsTs numRidesTs driversPmtTs
   where
+    makeGranular (Just timeVal) = makeSlidingWindowKey granuality "" timeVal
```

Comment:
Isn't this still on timestamp basis always?

Reply:
no like if the param is hourly then this function will group on the basis of hours , and same for daily.